x-pack/plugins/observability_solution/observability_ai_assistant_app/public/hooks/use_nav_control_screen_context.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+import datemath from '@elastic/datemath';
+import moment from 'moment';
+import { createScreenContextAction } from '@kbn/observability-ai-assistant-plugin/public';
+import { useKibana } from './use_kibana';
+import { useObservabilityAIAssistantAppService } from './use_observability_ai_assistant_app_service';
+
+export function useNavControlScreenContext() {
+  const service = useObservabilityAIAssistantAppService();
+
+  const {
+    services: {
+      plugins: {
+        start: { data },
+      },
+    },
+  } = useKibana();
+
+  const { from, to } = data.query.timefilter.timefilter.getTime();
+
+  const [href, setHref] = useState(window.location.href);
+
+  useEffect(() => {
+    const originalPushState = window.history.pushState.bind(window.history);
+    const originalReplaceState = window.history.replaceState.bind(window.history);
```

Comment:
it's hard to clean up because something else might have overridden `window.history.pushState` at that point. I'll add a flag for the `setState` calls and clean up the window listeners.